src/polyfills/object-assign.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Copies values of all enumerable own properties from one or more source
+ * objects (provided as extended arguments to the function) to a target object.
+ *
+ * @param {!Object} target
+ * @returns {!Object}
+ */
+export function assign(target) {
+  if (target == null) {
+    throw new TypeError('Cannot convert undefined or null to object');
+  }
+
+  const hasOwnProperty = Object.prototype.hasOwnProperty;
```

Comment:
Do you mean this should be defined outside the function? Just curious why is that better?